addon/mixins/flexberry-maplayer-actions-handler.js

```diff
@@ -364,5 +367,47 @@ export default Ember.Mixin.create({
 
     Ember.set(layer, 'isDeleted', true);
     return layer;
-  }
+  },
+
+  /**
+   Gets options for setIndexes method.
+
+    @method getIndexesOptions
+    @param {Object} layer Layer to get options for.
+    @returns {Object} First index and layers (options.firstIndex and options.layers).
+  */
+  getIndexesOptions(layer) {
```

Comment:
Recursion in this method is unnecessary. It could be simly transformed into loop.